MainAnalysis/20241122_CorrectionOPALComp/Correction.cpp

```diff
@@ -0,0 +1,1230 @@
+   
+   // now need to translate the thetas to zs
+   const int BinCount = 100;
+   double zBins[2*BinCount+1]; 
+   double zBinMin = (1- cos(0.002))/2; 
+   double zBinMax = 0.5;
+
+   for(int i = 0; i <= BinCount; i++){
+      zBins[i] = exp(log(zBinMin) + (log(zBinMax) - log(zBinMin)) / BinCount * i);
+      zBins[2*BinCount-i] = zBinMax * 2 - exp(log(zBinMin) + (log(zBinMax) - log(zBinMin)) / BinCount * i);
+   }
+
+   TH1D HTemp("HTemp", "HTemp", 200, 0, 200); 
+
+   TGraphAsymmErrors graph(99); 
+   
+   // first loop - fill histo
+   for(int i = 1; i < 99; i++)
+   {
+      int iGraph = i-1;
+      // get the central value for the data-point in degrees
+      double thetaDegrees = centralvals[iGraph][0]; 
+      // then convert it to radians
+      double thetaRadians = (TMath::Pi()*thetaDegrees)/180; 
+      // then get the z value
+      double zval = (1-cos(thetaRadians))/2; 
+      // find the bin in the histogram that this should go in so that it can be plotted correctly
+      int bin = FindBin(zval, 2*BinCount, zBins);
+      // find the fraction within the bin so that it is drawn at the correct point (i.e. bin width change does not matter)
+      double binFraction = FindBinFraction(zval, 2*BinCount, zBins); 
+      // get the sum val using the jacobian
+      double sumVal = centralvals[iGraph][1] * (2.0/sin(thetaRadians)); 
+      std::cout << "theta in degrees is " << thetaDegrees << " z value is " << zval << " bin " << bin  << " sumVal is " << sumVal << std::endl;
+      // set the correct point on the graph
+      graph.SetPoint(iGraph, HTemp.GetBinLowEdge(bin)+binFraction, sumVal);
```

Comment:
This seems like a bug!

The bin boundaries are as follows
Bin #0 lower bound is stored in zBins[0], upper bound is stored in zBins[1]
this is the assumption used in the FindBin and FindBinFraction.

Then...in root paradigm, the first bin is indexed 1.  So the bin corresponding to zBins[0] to zBins[1] is actually bin + 1.

This could be the problem of the OPAL shift.

Reply:
Ah I see, so you mean it should be `HTemp.GetBinLowEdge(bin-1)`?

Reply:
Ah yes, you're right - this makes the agreement in line with what is seen in the linear binning! thanks

Reply:
Cool!  Is this the latest code?  Shall I expect new commits before looking further and merge?

Reply:
There will be a commit to fix this bug and then I am trying to investigate the systematic uncertainties further so that may result in another commit (if I find an issue).

Reply:
Cool, so I'll wait until I get greenlight from you to start checking the code